gp_emulator/GaussianProcess.py

```diff
@@ -221,35 +223,113 @@ def predict ( self, testing, do_unc=True ):
 	do_unc: flag, optional
 		Calculate the uncertainty (if you don't set this flag, it
 		can shave a few us"""
-
+       
 
         ( nn, D ) = testing.shape
         assert D == self.D
-        
         expX = np.exp ( self.theta )
         
         a = dist.cdist ( np.sqrt(expX[:(self.D)])*self.inputs, \
             np.sqrt(expX[:(self.D)])*testing, 'sqeuclidean')
-        
         a = expX[self.D]*np.exp(-0.5*a)
         b = expX[self.D]
         
         mu = np.dot( a.T, self.invQt)
+
         if do_unc:
-	    var = b - np.sum (  a * np.dot(self.invQ,a), axis=0)
+	       var = b - np.sum (  a * np.dot(self.invQ,a), axis=0)
         # Derivative and partial derivatives of the function
         deriv = np.zeros ( ( nn, self.D ) )
 
         for d in xrange ( self.D ):
             aa = self.inputs[:,d].flatten()[None,:] - testing[:,d].flatten()[:,None]
             c = a*aa.T
-
             deriv[:, d] = expX[d]*np.dot(c.T, self.invQt)
         if do_unc:
             return mu, var, deriv
         else:
 	    return mu, deriv
         
+        
+    def gpu_predict ( self, testing, prec, threashold):# self, testing, do_unc=True):
```

Comment:
Threshold is spelled wrong.
Prefer "precision" to prec.
Don't leave commented out code in the files.